### PR TITLE
Remove all FLAnimatedImage from the Core SDWebImage repo. Move the code to SDWebImageFLPlugin repo

### PR DESCRIPTION
**Pull request by dreampiggy**
### New Pull Request Checklist

* [x] I have read and understood the [CONTRIBUTING guide](https://github.com/rs/SDWebImage/blob/master/.github/CONTRIBUTING.md)
* [x] I have read the [Documentation](http://cocoadocs.org/docsets/SDWebImage/)
* [x] I have searched for a similar pull request in the [project](https://github.com/rs/SDWebImage/pulls) and found none

* [x] I have updated this branch with the latest master to avoid conflicts (via merge from master or rebase)
* [x] I have added the required tests to prove the fix/feature I am adding
* [x] I have updated the documentation (if necessary)
* [x] I have run the tests and they pass
* [x] I have run the lint and it passes (`pod lib lint`)

This merge request fixes / reffers to the following issues: #2275 

### Pull Request Description

This PR, move all the `FLAnimatedImage` module integrate code from our SDWebImage main repo, to another standalone repo [SDWebImageFLPlugin](https://github.com/SDWebImage/SDWebImageFLPlugin) instead.

In SDWebImage 5.x, we does not need to rely on FLAnimatedImageView to support better GIF animation. We have our [Animated Image Solution](https://github.com/rs/SDWebImage/wiki/Advanced-Usage#animated-image-50), which provide a better performance, support all animated image format, easy to integrate for our framework and allow customization.

However, we should keep a way for our 4.x user who already use FLAnimatedImage deeply. So we provide a plugin for SDWebImage 5.x. The code which contribute to the master branch or the issue will also be transfered there. Let's start doing it.

### CODE REVIEW

File: Tests/Tests/SDWebCacheCategoriesTests.m [hunk 1]
```diff
@@ -118,25 +118,6 @@ - (void)testUIButtonSetBackgroundImageWithURLNormalState {
...
-                            expect(image).toNot.beNil();
-                            expect(error).to.beNil();
-                            expect(originalImageURL).to.equal(imageURL);
-                            
-                            expect(imageView.animatedImage).toNot.beNil();
-                            [expectation fulfill];
-                                }];
-    [self waitForExpectationsWithCommonTimeout];
-}
-#endif
```
Comment (bpoplauschi) [inline Tests/Tests/SDWebCacheCategoriesTests.m]: this `#endif` was accidentaly deleted and makes the build fail

**Comment by bpoplauschi** [comment]
@dreampiggy this looks good, except for a small CI issue which I am correcting now.
One thing I noticed, there are some differences between the `FLAnimatedImageView+WebCache` files from this repo and the ones from (SDWebImageFLPlugin)[https://github.com/SDWebImage/SDWebImageFLPlugin]. We should probably update this repo with the latest changes. 


> However, we should keep a way for our 4.x user who already use FLAnimatedImage deeply. So we provide a plugin for SDWebImage 5.x. The code which contribute to the master branch or the issue will also be transfered there. Let's start doing it.

I don't think we need to do anything here. The 4.x users still have the 4.x code with `FLAnimatedImage` support. 5.x users will just need to add `SDWebImageFLPlugin` into their `Podfile`.

File: Tests/Tests/SDWebCacheCategoriesTests.m (hunk 1 above)
Comment (bpoplauschi) [reply Tests/Tests/SDWebCacheCategoriesTests.m]: Fixed with c797279 commit on the `project_move_FLAnimatedImage` branch

**Comment by dreampiggy** [comment]
@bpoplauschi Thanks. CI now pass.

**Comment by bpoplauschi** [comment]
@dreampiggy I've imported the small changes from the main repo to the `SDWebImageFLPlugin` one: [b45963b](https://github.com/SDWebImage/SDWebImageFLPlugin/commit/b45963b607e7a6bd286114e332a39850804a11e2)

**Comment by bpoplauschi** [comment]
I'll go ahead and merge this PR.

**Comment by dreampiggy** [comment]
@bpoplauschi That SDWebImageFLPlugin seems not finished with current All FL code (Loss some commit after 5.x and does not contains a CI). I'll try to move and polish later...

**Comment by bpoplauschi** [comment]
@dreampiggy I just enabled Travis CI for `SDWebImageFLPlugin` and `SDWebImagePhotosPlugin`.
Indeed, we need to write the `Readme` for those projects.
What commits do you mean were lost?

**Comment by dreampiggy** [comment]
@bpoplauschi I added it. Some small change from #2378 which not synced after that code movemenet.

**Comment by bpoplauschi** [comment]
Got it. Thanks.

**Pull request merged**